scripts/docker-integration-tests/carbon/test.sh

```diff
@@ -29,6 +31,19 @@ function read_carbon {
   return $?
 }
 
+function find_carbon {
```

Comment:
nit: instead of a single `carbon` integration tests, mind refactoring this into two tests - one for `find-carbon` and another for `read-carbon`.

Reply:
Wouldn't that make integration runs another minute or so longer? (namespaces seem to take 40~ sec to come up)

Reply:
Fair. I don't like coupled tests but the extra 40s does seem wasteful.